sequencer/src/lib.rs

```diff
@@ -286,38 +290,46 @@ pub async fn init_node(
     let public_ip = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
     let orchestrator_client = OrchestratorClient::new(validator_args, public_ip.to_string());
 
-    let (mut config, wait_for_orchestrator) = match persistence.load_config().await? {
+    let (config, wait_for_orchestrator) = match persistence.load_config().await? {
         Some(config) => {
             tracing::info!("loaded network config from storage, rejoining existing network");
             (config, false)
         }
         None => {
             tracing::info!("loading network config from orchestrator");
-            let config = orchestrator_client.get_config(public_ip.to_string()).await;
+            let mut config: NetworkConfig<VerKey, StaticElectionConfig> =
+                orchestrator_client.get_config(public_ip.to_string()).await;
+
+            config = orchestrator_client
+                .post_and_wait_all_public_keys(
+                    config.node_index,
+                    config.config.my_own_validator_config.public_key,
+                )
+                .await;
             tracing::info!("loaded config, we are node {}", config.node_index);
             persistence.save_config(&config).await?;
             (config, true)
         }
     };
     let node_index = config.node_index;
 
-    // Generate public keys and this node's private keys.
+    // Generate node's private keys.
     //
     // These are deterministic keys suitable *only* for testing and demo purposes.
-    config.config.my_own_validator_config =
-        ValidatorConfig::<PubKey>::generated_from_seed_indexed(config.seed, node_index, 1);
+
     let priv_key = config.config.my_own_validator_config.private_key.clone();
     let num_nodes = config.config.total_nodes.get();
-    let pub_keys = (0..num_nodes)
-        .map(|i| PubKey::generated_from_seed_indexed(config.seed, i as u64).0)
+
+    let known_nodes_with_stake: Vec<<PubKey as SignatureKey>::StakeTableEntry> =
+        config.config.known_nodes_with_stake.clone();
+    let pub_keys = known_nodes_with_stake
+        .iter()
+        .map(|entry| entry.stake_key)
         .collect::<Vec<_>>();
-    let known_nodes_with_stake: Vec<<PubKey as SignatureKey>::StakeTableEntry> = (0..num_nodes)
-        .map(|id| pub_keys[id].get_stake_table_entry(1u64))
-        .collect();
     let state_ver_keys = (0..num_nodes)
         .map(|i| StateKeyPair::generate_from_seed_indexed(config.seed, i as u64).ver_key())
         .collect::<Vec<_>>();
```

Comment:
We now have the same problem with the state signing schnorr keys, where they have to be generated deterministically so that we know all the public keys. So we may have to do something similar using the orchestrator to get these. @dailinsubjam @bfish713 maybe we can add `state_pub_key` to `StakeTableEntry`, and have nodes post both their public staking key and public schnorr key?